runtime/runtime.c

```diff
@@ -87,7 +88,13 @@ INLINE i32 i32_div(i32 a, i32 b) {
 }
 
 INLINE i32 i32_rem(i32 a, i32 b) {
-    awsm_assert(b && (a != INT32_MIN || b != -1));
+    awsm_assert(b != 0);
+
+    /* Because MIN is one less than MAX, we can FPE here */
+    if (unlikely(a == INT32_MIN)) {
+        return (a + abs(b)) % b;
```

Comment:
This is hand written by me. Based on https://github.com/sunfishcode/wasm-reference-manual/blob/master/WebAssembly.md#integer-remainder-signed, the wasm spec differs from modulo operations. There is an llvm `srem` intrinsic, but it seems to track C and have undefined behavior with INT32_MIN and -1.

My logic is just that since division/multiplication can be performed by repeated subtraction/addition, we can add the absolute value of the denominator to be sure that the numerator doesn't overflow, and that results in a numerator between {INT32_MIN, 0}. Since we can't produce a positive number, I don't think that should cause any sign issues. 

Here's a REPL with some experiments:
https://replit.com/@spmcbride1201/i32rem#main.c

Perhaps we might want to enable to a "quirks mode" where we drop strict compliance for the wasm spec in certain cases? Sort of like how bash takes a flag for strict POSIX shell standards compliance. I suggest we kick that discussion to an issue. It seems like a good idea to try to target strict compliance first and then consciously downshift if there are perf wins for doing so.